Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -0,0 +1,172 @@
+
+
+Optionally, Egress Gateway support can be enabled by setting
+``enable-egress-gateway`` to ``true`` in ConfigMap. By default, it is disabled.
+
+To set ``enable-egress-gateway`` to ``true``, it requires
+``enable-bpf-masquerade`` and ``kube-proxy-replacement``. It cannot be enabled
+with either of these 2 options set as ``false`` or ``disabled``.
+
+Architecture
+============
+
+Overall, some nodes in the cluster will be assigned as gateway nodes. These gateway nodes not
+only have a node IP, but also have additional pools of egress IPs attached to it. These egress IP
+addresses can be associated with certain workloads (pods) that run in the cluster. When certain
+packets are selected, instead of exiting the cluster immediately, they will be forwarded to a gateway
+node and SNAT-ed with an egress IP there. A tunnel interface will be created when Egress Gateway is
+enabled. The tunnel is for forwarding with Egress Gateway. All other traffic will keep their
+own behavior based on which tunnel mode is selected.
+
+The following 2 diagrams show the packet flow for both egress and return traffic when using direct
+routing mode. Note that the return packet is not going over a tunnel, while forwarding traffic is. 
+
+.. image:: images/cilium_egress_gateway_flow.png
+   :scale: 70%
+
+.. image:: images/cilium_egress_gateway_return_flow.png
+   :scale: 70%
+
+
+
+Create an External Service
+==========================
+
+Create an nginx service on a Linux node that is external to the existing Kubernetes
+cluster, and use it as the destination of the egress traffic.
+
+.. code-block:: shell-session
+
+    $ # Install and start nginx
+    $ sudo apt install nginx
+    $ sudo systemctl start nginx
+
+    $ # Make sure the service is started and listens on port :80
+    $ sudo systemctl status nginx
+    ● nginx.service - A high performance web server and a reverse proxy server
+    Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
+    Active: active (running) since Sun 2021-04-04 21:58:57 UTC; 1min 3s ago
+    [...]
+    $ curl http://192.168.33.13:80  # Assume 192.168.33.13 is the external IP of the node
+    [...]
+    <title>Welcome to nginx!</title>
+    [...]
+
+Create Client Pods
+==================
+
+Deploy a client pod from which egress traffic is generated, and try to access previous
+external service.
+
+.. code-block:: shell-session
+
+    $ kubectl run -it client --image=busybox
+    If you don't see a command prompt, try pressing enter.
+    / # wget http://192.168.33.13:80 -o /dev/null
+
+Verify access log from previous node that the request is coming from one of the node in
+Kubernetes cluster.
+
+.. code-block:: shell-session
+
+    $ tail /var/log/nginx/access.log
+    192.168.33.11 - - [04/Apr/2021:22:06:57 +0000] "GET / HTTP/1.1" 200 612 "-" "Wget/1.19.4 (linux-gnu)"
+
+In the previous example, the busybox is running on the node ``192.168.33.11``, so the result makes sense.
```

Comment:
I think we can skip this first check. This is essentially testing Cilium works, which is out of scope for the guide IMO. If you really want to show before/after, I would configure everything, show before, add the label, show after.